doc: update nRF9160 user guide with VSC and command line instructions (#7986)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-changelog.rst

  * :ref:`ug_thread_configuring` to better indicate what is required and what is optional.
    Also added further clarifications to the page to make everything more clear.
  * :ref:`ug_nrf9160_gs` by moving :ref:`nrf9160_gs_updating_fw_modem` before :ref:`nrf9160_gs_updating_fw_application` because updating modem firmware erases application firmware.

>   * :ref:`build_pgm_nrf9160` section with |VSC| and command line instructions on the :ref:`ug_nrf9160` documentation.

Reviewer: ```suggestion
  * :ref:`build_pgm_nrf9160` section in the :ref:`ug_nrf9160` documentation by adding |VSC| and command line instructions.
```
